# JOINT STATE CONTROLLER
joint_state_controller:
  type: joint_state_controller/JointStateController
  publish_rate: 50
# TORSO + RIGHT ARM TRAJECTORY CONTROLLER + LEFT ARM TRAJECTORY CONTROLLER
rcta_joint_trajectory_controller:
  type: position_controllers/JointTrajectoryController
  joints:
    - torso_joint1
    - limb_right_joint1
    - limb_right_joint2
    - limb_right_joint3
    - limb_right_joint4
    - limb_right_joint5
    - limb_right_joint6
    - limb_right_joint7
    - limb_left_joint1
    - limb_left_joint2
    - limb_left_joint3
    - limb_left_joint4
    - limb_left_joint5
    - limb_left_joint6
    - limb_left_joint7
constraints:
  goal_time: 0.5
  stopped_velocity_tolerance: 0.02
  #allow_partial_joints_goal: True
rcta_right_robotiq_controller:
  type: position_controllers/JointGroupPositionController
  joints:
    - limb_right_finger_1_joint_1
    - limb_right_finger_2_joint_1
    - limb_right_finger_middle_joint_1
rcta_left_robotiq_controller:
  type: position_controllers/JointGroupPositionController
  joints:
    - limb_left_finger_1_joint_1
    - limb_left_finger_2_joint_1
    - limb_left_finger_middle_joint_1
rcta_velocity_controller:
  type: "diff_drive_controller/DiffDriveController"
  left_wheel: ['front_left_wheel', 'rear_left_wheel']
  right_wheel: ['front_right_wheel', 'rear_right_wheel']
  pose_covariance_diagonal: [0.001, 0.001, 0.001, 0.001, 0.001, 0.03]
  twist_covariance_diagonal: [0.001, 0.001, 0.001, 0.001, 0.001, 0.03]
  pose_covariance_diagonal: [0.001, 0.001, 1000000.0, 1000000.0, 1000000.0, 0.03]
  twist_covariance_diagonal: [0.001, 0.001, 0.001, 1000000.0, 1000000.0, 0.03]
  cmd_vel_timeout: 0.25
